src/explore-education-statistics-common/src/modules/table-tool/components/TimePeriodDataTable.tsx

```diff
@@ -112,6 +112,14 @@ const TimePeriodDataTableInternal = forwardRef<HTMLElement, Props>(
   ) {
     const { subjectMeta, results } = fullTable;
 
+    if (!tableHeadersConfig.rows.length) {
+      return (
+        <WarningMessage>
+          There was a problem rendering the table.
+        </WarningMessage>
```

Comment:
Would suggest adding better monitoring whenever a broken table might render. Doing this via Google Analytics would probably be the best place to start (we could add Application Insights tracking later if we think it might be useful). We could create a re-usable component like this:

```tsx
const LogGAEvent = (event: AnalyticsEvent) => {
  useEffect(() => {
    googleAnalyticsService.logEvent(event);
    // Probably a good idea to add in some
    // other stuff to stop logging on every render (you can figure that out)
  }, [event]);

  return null;
}
```

Might want to check with Cam about what the best categorisation/labelling for this might be.

We could then use this component in any of the places where we render `WarningMessage` in `TimePeriodDataTable` (this includes the error boundary) like:

```tsx
<>
  <LogGAEvent ...theEvent />
  <WarningMessage>
    There was a problem rendering the table.
  </WarningMessage>
</>
```